react_excercise/got-app/src/Api.js

```diff
@@ -0,0 +1,80 @@
+import axios from 'axios';
+import levenshteinDistance from './LevenshteinDistance';
+
+const url = 'https://thronesapi.com/api/v2/Characters';
+
+async function fetchData() {
+    try {
+        const response = await axios.get(url);
+        return response.data;
+    } catch (error) {
+        console.error('Error fetching data from API');
+        throw error;
+    }
+}
+
+async function searchForCharacters(input) {
+    try {
+        const results = await fetchData();
```

Comment:
Minor - We know that api data is small, so we can cache it in memory for better UX. Also making and responding to api requests is wasteful for both client and server.

Reply:
I'll look into this more.  Thanks!